airbyte-integrations/connectors/source-greenhouse/source_greenhouse/client.py

```diff
@@ -67,16 +68,42 @@ def list(self, name, **kwargs):
     def _enumerate_methods(self) -> Mapping[str, callable]:
         return {entity: partial(self.list, name=entity) for entity in self.ENTITIES}
 
+    def get_accessible_endpoints(self) -> List[str]:
+        """Try to read each supported endpoint and return accessible stream names"""
+        logger = AirbyteLogger()
+        accessible_endpoints = []
+        for entity in self.ENTITIES:
+            try:
+                getattr(self._client, entity).get()
+                accessible_endpoints.append(entity)
+            except HTTPError as error:
+                logger.warn(f"Endpoint '{entity}' error: {str(error)}")
+                if "This API Key does not have permission for this endpoint" not in str(error):
```

Comment:
Are there any codes along with this message? If so, probably it's better to use the code instead of the message.

Reply:
code is 403, but I would keep relying on the error message to match this particular error.